lib/charms/mongodb/v1/mongodb_tls.py

```diff
@@ -302,21 +315,93 @@ def _on_certificate_expiring(self, event: CertificateExpiringEvent) -> None:
 
         self.set_tls_secret(internal, Config.TLS.SECRET_CSR_LABEL, new_csr.decode("utf-8"))
 
-    def _get_sans(self) -> List[str]:
+    def get_new_sans(self) -> Dict:
         """Create a list of DNS names for a MongoDB unit.
 
         Returns:
             A list representing the hostnames of the MongoDB unit.
         """
+        sans = {}
+
         unit_id = self.charm.unit.name.split("/")[1]
-        return [
+        sans[SANS_DNS_KEY] = [
             f"{self.charm.app.name}-{unit_id}",
             socket.getfqdn(),
-            f"{self.charm.app.name}-{unit_id}.{self.charm.app.name}-endpoints",
-            str(self.charm.model.get_binding(self.peer_relation).network.bind_address),
             "localhost",
+            f"{self.charm.app.name}-{unit_id}.{self.charm.app.name}-endpoints",
+        ]
+
+        sans[SANS_IPS_KEY] = [
+            str(self.charm.model.get_binding(self.peer_relation).network.bind_address)
         ]
 
+        if self.charm.is_role(Config.Role.MONGOS) and self.charm.is_external_client:
+            sans[SANS_IPS_KEY].append(
+                self.charm.get_ext_mongos_host(self.charm.unit, incl_port=False)
+            )
+
+        return sans
+
+    def get_current_sans(self, internal: bool) -> List[str] | None:
+        """Gets the current SANs for the unit cert."""
+        # if unit has no certificates do not proceed.
+        if not self.is_tls_enabled(internal=internal):
+            return
+
+        try:
+            sans_lines = self.get_sans_from_host(internal)
+        except (subprocess.CalledProcessError, ExecError) as e:
+            logger.error("failed to get sans from host, error: %s", e.stderr)
+            raise e
+
+        line = ""
+        for line in sans_lines:
+            if "DNS" in line or "IP" in line:
+                break
+
+        sans_ip = []
+        sans_dns = []
+        for item in line.split(", "):
```

Comment:
should this block be in the above `for` loop? otherwise, when is `line` set in the previous loop?